callouts/go/extproc/examples/client/client.go

```diff
@@ -0,0 +1,197 @@
+)
+
+// makeChannel creates a gRPC client connection to the given address.
+func makeChannel(addr string, useTLS bool, certFile string) (*grpc.ClientConn, error) {
+	if useTLS {
+		creds, err := credentials.NewClientTLSFromFile(certFile, "")
+		if err != nil {
+			return nil, err
+		}
+		return grpc.Dial(addr, grpc.WithTransportCredentials(creds))
+	}
+	return grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+}
+
+// makeJSONRequest sends ProcessingRequest objects read from a JSON string and receives ProcessingResponse objects.
+func makeJSONRequest(address string, useTLS bool, certFile, jsonData string) ([]*extproc.ProcessingResponse, error) {
+	conn, err := makeChannel(address, useTLS, certFile)
+	if err != nil {
+		return nil, err
+	}
+	defer func(conn *grpc.ClientConn) {
+		err := conn.Close()
+		if err != nil {
+
+		}
+	}(conn)
+
+	client := extproc.NewExternalProcessorClient(conn)
+
+	// Unmarshal the JSON string into a slice of ProcessingRequest
+	var rawRequests []map[string]interface{}
+	if err := json.Unmarshal([]byte(jsonData), &rawRequests); err != nil {
+		return nil, err
+	}
+
+	var requests []*extproc.ProcessingRequest
+	for _, rawReq := range rawRequests {
+		reqBytes, err := json.Marshal(rawReq)
+		if err != nil {
+			return nil, err
+		}
+		req := &extproc.ProcessingRequest{}
+		if err := protojson.Unmarshal(reqBytes, req); err != nil {
+			return nil, err
+		}
+		requests = append(requests, req)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	stream, err := client.Process(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, req := range requests {
+		if err := stream.Send(req); err != nil {
+			return nil, err
+		}
+		log.Printf("Sent request: %v", req)
+	}
+
+	// Close the send direction of the stream to signal the server that all messages have been sent.
+	if err := stream.CloseSend(); err != nil {
+		return nil, err
+	}
+
+	var responses []*extproc.ProcessingResponse
+	for {
+		resp, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		responses = append(responses, resp)
+		log.Printf("Received response: %v", resp)
+	}
+
+	return responses, nil
+}
+
+func main() {
+	flag.Parse()
+
+	if *dataJSON == "" {
+		log.Fatal("Data JSON is required")
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	// Start the server in a goroutine
+	go startServer(&wg, *serverAddr)
+
+	// Wait for the server to start
+	wg.Wait()
+	time.Sleep(3 * time.Second) // Ensure the server has enough time to start
+
+	responses, err := makeJSONRequest(*serverAddr, *tls, *certFile, *dataJSON)
+	if err != nil {
+		log.Fatalf("Failed to make JSON request: %v", err)
+	}
+
+	for _, resp := range responses {
+		respJSON, err := protojson.Marshal(resp)
+		if err != nil {
+			log.Fatalf("Failed to marshal response to JSON: %v", err)
+		}
+		log.Printf("Response: %s", string(respJSON))
```

Comment:
Did I understand correctly that the purpose of this part is to print responses pretty? If yes, could we use [`prototext.Format`](https://pkg.go.dev/google.golang.org/protobuf/encoding/prototext)?